File: 0000xx/000002/ListNode.java
public class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }

    public static ListNode toListNode(int[] intAry){
        if(intAry==null)return null;
        if(intAry.length==0)return null;
        ListNode ret = null;
        for(int i=intAry.length-1;i>=0;--i){
            ListNode old = ret;
            ret = new ListNode(intAry[i]);
            ret.next = old;
        }
        return ret;
    }
    
    public static int[] toIntArray(ListNode head){
        if(head==null)return new int[0];
        int len=0;
        for(ListNode i = head;i!=null;i=i.next)++len;
        
        int[] ret = new int[len];
        int outIdx = 0;
        for(ListNode i = head;i!=null;i=i.next){
            ret[outIdx]=i.val;
            ++outIdx;
        }
        
        return ret;
    }
}

File: 0010xx/001019/kotlin/ListNode.kt
class ListNode(var `val`: Int) {
  var next: ListNode? = null
}

fun toStr(head: ListNode?): String{
  val sb=StringBuilder()
  sb.append("[")
  var isFirst = false
  var ptr = head
  while(ptr!=null){
    if(!isFirst){sb.append(",")}
    sb.append(ptr.`val`.toString())
    ptr = ptr.next
  }
  sb.append("]")
  return sb.toString()  
}

fun toListNode(str: String): ListNode?{
  val vIntQAry = toIntQAryQ(str.toCharArray(),0).second!!
  return toListNode(vIntQAry)
}

fun toListNode(intAry: IntArray):ListNode?{
  if(intAry.size==0){return null}
  var ret:ListNode? = null
  for(i in intAry.indices.reversed()){
    //println("i=%d".format(i))
    var prev = ListNode(intAry[i])
    prev.next = ret
    ret = prev
  }
  return ret
}

fun toListNode(intAry: Array<Int?>):ListNode?{
  if(intAry.size==0){return null}
  var ret:ListNode? = null
  for(i in intAry.indices.reversed()){
    //println("i=%d".format(i))
    var prev = ListNode(intAry[i]!!)
    prev.next = ret
    ret = prev
  }
  return ret
}
